src/noise/quantum_error.hpp

```diff
@@ -350,13 +372,22 @@ const std::vector<cmatrix_t>& QuantumError::kraus() const {
   return canonical_kraus_;
 }
 
-
 void QuantumError::compute_superoperator() {
+#if defined(_MSC_VER) || defined(GNUC_AVX2)
```

Comment:
The ifdefs seem a bit messy here, can't it just be like the controller class with:
```
if (is_avx2_supported()) {
  compute_superoperator_<QubitSuperoperator::State<QV::SuperoperatorAvx2<double>>>();
} else {
  compute_superoperator_<QubitSuperoperator::State<QV::Superoperator<double>>>();
}
```